src/main/java/io/strimzi/kafka/metrics/KafkaMetricsCollector.java

```diff
@@ -109,24 +108,28 @@ String metricName(MetricName metricName) {
         return prefix + '_' + group + '_' + name;
     }
 
-    static MetricFamilySamples convert(String name, String help, KafkaMetric metric, Map<String, String> labels) {
-        Object value = metric.metricValue();
-        if (!(value instanceof Number)) {
-            // Prometheus only accepts numeric metrics.
-            // Kafka gauges can have arbitrary types, so skip them for now
-            // TODO move non-numeric values to labels
-            return null;
-        }
+    static MetricFamilySamples convert(String name, String help, KafkaMetric metric, Map<String, String> labels, MetricName metricName) {
         Map<String, String> sanitizedLabels = labels.entrySet().stream()
                 .collect(Collectors.toMap(
                         e -> Collector.sanitizeMetricName(e.getKey()),
                         Map.Entry::getValue,
                         (v1, v2) -> {
-                            throw new IllegalStateException("Unexpected duplicate key " + v1);
+                            LOG.warn("Unexpected duplicate key" +  v1);
```

Comment:
Always use patterns when logging.  I'd suggest adding the word Ignoring as it makes it clear from the logs what the code is doing with the duplicate.

```suggestion
                            LOG.warn("Ignoring metric value duplicate key {}", v1);
```